pkg/api/short_urls.go

```diff
@@ -14,15 +15,16 @@ import (
 func (hs *HTTPServer) createShortURL(c *models.ReqContext, cmd dtos.CreateShortURLForm) Response {
 	hs.log.Debug("Received request to create short URL", "path", cmd.Path)
 
-	uid, err := hs.ShortURLService.CreateShortURL(c.SignedInUser, strings.TrimPrefix(cmd.Path, "/"))
+	uid, err := hs.ShortURLService.CreateShortURL(c.SignedInUser, strings.TrimPrefix(cmd.Path, setting.AppUrl))
 	if err != nil {
 		c.Logger.Error("Failed to create short URL", "error", err)
 		return Error(500, "Failed to create short URL", err)
 	}
 
-	c.Logger.Debug("Created short URL", "uid", uid)
+	shortURL := fmt.Sprintf("%sgoto/%s", setting.AppUrl, uid)
```

Comment:
Shouldn't we make sure here that there's a slash before "goto"? This code assumes that setting.AppUrl ends with a slash, which could quickly go wrong. Maybe it already doesn't work, I don't even know?

I think the following is safer:

```suggestion
	shortURL := path.Join(setting.AppUrl, "goto", uid)
```

Reply:
Yes, totally makes sense! Good catch.